speechbrain/lobes/models/huggingface_wav2vec.py

```diff
@@ -39,162 +40,445 @@
-    """This lobe enables the integration of HuggingFace and SpeechBrain
-    pretrained wav2vec2.0/Hubert models.
+    # Check if source is downloaded already
+    sink = pathlib.Path(
+        save_path + "/models--" + path.replace("/", "--") + "/snapshots"
+    )
+    if sink.exists():
+        sink = sink / os.listdir(str(sink))[0]  # there's a hash-id subfolder
+        if any(
+            File.endswith(".bin") or File.endswith(".ckpt")
+            for File in os.listdir(str(sink))
+        ):
+            is_local = True
+            local_path = str(sink)
+        else:
+            local_path = path
+    else:
+        local_path = path
+
+    if is_local:
+        # Test for HuggingFace model
+        if any(File.endswith(".bin") for File in os.listdir(local_path)):
+            is_sb = False
+            return is_sb, checkpoint_filename
+
+        # Test for SpeechBrain model and get the filename.
+        for File in os.listdir(local_path):
+            if File.endswith(".ckpt"):
+                checkpoint_filename = os.path.join(path, File)
+                is_sb = True
+                return is_sb, checkpoint_filename
+    else:
+        files = model_info(path).siblings  # get the list of files of the Hub
+
+        # Test if it's an HuggingFace model or a SB one
+        for File in files:
+            if File.rfilename.endswith(".ckpt"):
+                checkpoint_filename = File.rfilename
+                is_sb = True
+                return is_sb, checkpoint_filename
 
-    Source paper wav2vec2.0: https://arxiv.org/abs/2006.11477
-    Source paper Hubert: https://arxiv.org/abs/2106.07447
-    Transformer from HuggingFace needs to be installed:
-    https://huggingface.co/transformers/installation.html
+        for File in files:
+            if File.rfilename.endswith(".bin"):
+                checkpoint_filename = File.rfilename
+                is_sb = False
+                return is_sb, checkpoint_filename
 
-    The model can be used as a fixed feature extractor or can be finetuned. It
-    will download automatically the model from HuggingFace or use a local path.
+    err_msg = f"{path} does not contain a .bin or .ckpt checkpoint !"
+    raise FileNotFoundError(err_msg)
+
+
+def config_return_hidden_states(config):
+    """Sets `output_hidden_states = True` for a transformer config.
+    """
+    config.output_hidden_states = True  # We want the hidden states as well!
+    return config
+
+
+def model_set_spectral_augmentation(model, apply_spec_augment):
+    """Sets `model.config.apply_spec_augment` the flag to a specific value.
+    """
+    model.config.apply_spec_augment = apply_spec_augment
+    return model
+
+
+def modify_state_dict_wav2vec2(path):
+    """Function to be used in HuggingFaceModel._load_sb_pretrained_parameters.
+
+    A custom loading ensures SpeechBrain compatibility for Pretrain and model
+    de/serialization. Here, '.wav2vec2' is removed before loading.
```

Comment:
do you think this problem can happen with other HF models? If so, could we instead have something that is more high-level? E.g the function takes a string in input that is removed from the key of the model (e.g, here it would be "model.wav2vec2."

Reply:
It should be provided by the user.
Maybe that needs to come out more clearly. This is just an example for how others can tackle it in their functions. This function is tasked only with wav2vec2 - we are not engineering around HF interfaces, as these revealed a moving target.

Reply:
OK. I see. 

I reread the python file and it makes sense.